CreeDictionary/CreeDictionary/views.py

```diff
@@ -86,18 +103,55 @@ def search_results(request, query_string: str):  # pragma: no cover
     )
 
 
-def lemma_details_internal(request, lemma_id: int):  # pragma: no cover
+@require_GET
+def lemma_details_internal(request):
     """
-    render paradigm table for a lemma
+    Render paradigm.html for a lemma. `index` view function renders a whole page that contains paradigm.html too.
+    This function, however, is used by javascript to dynamically replace the paradigm with the ones of different sizes.
+
+    `lemma-id` and `paradigm-size` are the expected query params in the request
+
+    4xx errors will have a single string as error message
+
+    :raise 400 Bad Request: when the query params are not as expected or inappropriate
+    :raise 404 Not Found: when the lemma-id isn't found in the database
+    :raise 405 Method Not Allowed: when method other than GET is used
     """
-    lemma = Wordform.objects.get(id=lemma_id)
+    lemma_id = request.GET.get("lemma-id")
+    paradigm_size = request.GET.get("paradigm-size")
+    # guards
+    if lemma_id is None or paradigm_size is None:
+        return HttpResponseBadRequest("query params missing")
+    try:
+        assert float(lemma_id) == int(lemma_id)
+        lemma_id = int(lemma_id)
+    except (ValueError, AssertionError):
```

Comment:
So, even though I would never knowingly turn off assertions, the `assert` statement CAN be turned off (which is dumb, but still). See https://dbader.org/blog/python-assert-tutorial#:~:text=Caveat%20#1

I honestly have no idea what your `assert` is trying to do here.

Reply:
this assert rule out the case when lemma-id is a float. if lemma-id is a float, without the assert. int() will cast the float to a integer.

Reply:
I tried it, and it just crashes :joy:

```python
>>> n = "3.14"
>>> float(n) == int(n)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '3.14'
```

Please remove the assert!

Reply:
aaah somehow I recalled wrongly int will cast😔